pkg/dag/packages.go

```diff
@@ -0,0 +1,222 @@
+package dag
+
+import (
+	"fmt"
+	"io/fs"
+	"log"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"chainguard.dev/melange/pkg/build"
+)
+
+// Configuration represents a configuration along with the file that sourced it
+type Configuration struct {
+	*build.Configuration
+	Path string
+}
+
+type Packages struct {
```

Comment:
This seems like an important concept in the overhauled design, so it should have a good doc comment.

Reply:
Added

Reply:
Awesome comment!